content/generic.rst

```diff
@@ -260,6 +260,28 @@ Each ID should be raised from a single place.
 The data attribute is a dictionary with key / values that make sense for the
 error.
 
+Don't use the `assert` statement in code but rather raise an explicit error.
+`assert` statement optimization is useless as it was scientifically proven
+that some bugs only show up when a customer uses the machine and we want
+those exceptions to be raised in production and not be accidentally
+disabled.
+
+Don't raise `AssertionError` outside of the test code.
+Use `RuntimeError` or a more specific exception.
+The `AssertionError` should never be handled nor in production nor in
+testing code.
+Using `self.assertRaises(AssertionError)` is a form of handling an
```

Comment:
I would place this and the above statement in their own paragraph; taking this into account I find it hard to understand when will we use them at all:

```
Don't raise `AssertionError` outside of the test code. [...] The `AssertionError` should never be handled [...] in testing code. Using `self.assertRaises(AssertionError)` is a form of handling an exception.
```